Helper/Order.php

```diff
@@ -1402,11 +1407,12 @@ protected function hasSamePrice($order, $transaction)
      * @param OrderModel $order
      * @throws \Exception
      */
-    protected function deleteOrder($order)
+    public function deleteOrder($order)
     {
         $this->eventsForThirdPartyModules->dispatchEvent("beforeFailedPaymentOrderSave", $order);
         try {
-            $order->cancel()->save()->delete();
+            $order->cancel()->save();
+            $this->orderRepository->delete($order);
```

Comment:
When we delete order via repository function it cleans cache in repository object